rasa/shared/core/events.py

```diff
@@ -314,21 +317,30 @@ def apply_to(self, tracker: "DialogueStateTracker") -> None:
 
     def __eq__(self, other: Any) -> bool:
         """Compares object with other object."""
-        if not isinstance(other, self.__class__):
-            return NotImplemented
-
-        return True
+        # Every class should implement this
```

Comment:
Does this mean that `Event` should be an abstract class? (And this an abstract method?)

Reply:
good catch! Should have been an abstract class all along